### PR TITLE
[Catalog] make all the fields required in the RJSF form while submitting the catalog request

### PR DESCRIPTION
**Pull request by Abhishek-kumar09**
Signed-off-by: Abhishek kr <abhimait1909@gmail.com>**Notes for Reviewers**

This PR fixes #

https://github.com/meshery/meshery/assets/48255244/b87c091d-e74d-486d-87d6-dc30aad0008a

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
In the next PR, hopefully, we can move to our standard modal.

**Comment by Abhishek-kumar09** [comment]
Oh that, I can do it now.

**Comment by leecalcote** [comment]
fixes #7865

File: ui/components/PublishModal.js [hunk 1]
```diff
@@ -40,15 +39,20 @@ export default function PublishModal(props) {
         "type" : "array",
         "items" : {
           "enum" : [
+            "Kubernetes",
+            "Argo CD",
+            "AWS App Mesh",
+            "Consul",
+            "Fluentd",
```
Comment (Avi-88) [inline ui/components/PublishModal.js:46]: Fluentd is named as fluentd-operator in the meshmodels fs structure. Also is app-mesh in the meshmodel the same as AWS App mesh here ? // @leecalcote

Comment (leecalcote) [reply ui/components/PublishModal.js:46]: @Avi-88 I think you have the right initial perspective: for lack of a better solution, let’s use the MeshModel model names as the sources of truth (use the MeshModel filesystem as the source).

**Comment by leecalcote** [comment]
…both here and in Meshery Cloud

**Comment by leecalcote** [comment]
Whether we have the complete and correct list or not, let's ensure this PR is merged before the next Meshery release. @MUzairS15 @Abhishek-kumar09 @Avi-88

**Pull request merged**